src/meili/routing.cc

```diff
@@ -162,7 +162,8 @@ inline bool IsEdgeAllowed(const baldr::DirectedEdge* edge,
                           const sif::cost_ptr_t& costing,
                           const Label& pred_edgelabel,
                           const baldr::GraphTile* tile) {
-  return !pred_edgelabel.edgeid().Is_Valid() || edgeid == pred_edgelabel.edgeid() ||
+  return (!pred_edgelabel.edgeid().Is_Valid() && costing->GetEdgeFilter()(edge) != 0.f) ||
```

Comment:
this is the important bug fix, previously this say, whatever edge the routing sees, its allowed so long as its at the beginning of the route. this allowed the route to actually start and end on an inaccesable edge and give a route the wrong way on a one way. now if we dont have a predecessor then we'll grab the edge filter and make sure the edge is at least allowed regardless of predecessor.